### PR TITLE
[MNT-19456] get nodes to cascade parameters fixed + test

### PR DESCRIPTION
**Pull request by alessandrobenedetti**
Cascading tracking fixed and test added.
The test uses a broken test SolrClient-API client so it was green even if the bug was there.
The fix of the client is in a separate Pull Request:  
https://github.com/Alfresco/alfresco-solrclient/pull/12

### CODE REVIEW

File: alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.util.LuceneTestCase;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.alfresco.solr.AlfrescoSolrUtils.*;
```
Comment (tpage-alfresco) [inline alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: Static imports should be first, and I don't think we usually use wildcard imports.  (IIRC There's a edge case issue that could cause a library upgrade to add additional classes that override our own ones.)

File: alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        params.add("start", "0");
+        params.add("rows", "6");
+        params.add("sort", "id asc");
+        params.add("fq", "{!afts}AUTHORITY_FILTER_FROM_JSON");
+        SolrServletRequest req = areq(params,
+            "{\"locales\":[\"en\"], \"templates\": [{\"name\":\"t1\", \"template\":\"%cm:content\"}], \"authorities\": [ \"mike\"], \"tenants\": [ \"\" ]}");
+        assertQ(req, "*[count(//doc)=2]", "//result/doc[1]/long[@name='DBID'][.='" + childFolderNode.getId() + "']", "//result/doc[2]/long[@name='DBID'][.='" + fileNode.getId() + "']");
+    }
+
+    private Acl initAcls() throws Exception
```
Comment (tpage-alfresco) [inline alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: Please could we have some Javadoc for this method.  What ACLs specifically does it create?

File: alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        ModifiableSolrParams params = new ModifiableSolrParams();
+        params.add("q", QueryConstants.FIELD_ANCESTOR + ":\"" + nodeRef.toString() + "\"");
+        params.add("qt", "/afts");
+        params.add("start", "0");
+        params.add("rows", "6");
+        params.add("sort", "id asc");
+        params.add("fq", "{!afts}AUTHORITY_FILTER_FROM_JSON");
+        SolrServletRequest req = areq(params,
+            "{\"locales\":[\"en\"], \"templates\": [{\"name\":\"t1\", \"template\":\"%cm:content\"}], \"authorities\": [ \"mike\"], \"tenants\": [ \"\" ]}");
+        assertQ(req, "*[count(//doc)=2]", "//result/doc[1]/long[@name='DBID'][.='" + childFolderNode.getId() + "']", "//result/doc[2]/long[@name='DBID'][.='" + fileNode.getId() + "']");
```
Comment (tpage-alfresco) [inline alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: This test seems very long, and I can't easily see what it's doing.  At what point do we change from setting up data to verifying things?  Where is the method under test?  Could the test be split into several shorter tests?

Comment (tpage-alfresco) [reply alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: Maybe it might help to have a Javadoc description at the top to explain what the test contains?  I see that we have some fairly good in-line comments to explain each block of code, but I didn't understand the overall aim of the test when reading it.

Comment (agazzarini) [reply alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: I restructured the test, moving initial setup in a @Before method, and isolating common  parts in  some compose method. Please have a look and let me know if it is ok

File: alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    {
+        initAlfrescoCore("schema.xml");
+    }
+
+    /**
+     * Setup initial data, which consists of the following hierarchy:
+     *
+     * |- a top level folder
+     * |---- a child folder
+     * |------ a file
```
Comment (tpage-alfresco) [inline alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: Great - thanks. This is exactly the kind of summary I was after!

Comment (tpage-alfresco) [reply alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java]: ... and give Mike access to it?
